### PR TITLE
Do not use "*" as DockerImageReference.Name for ImageStreamImport

### LINKED ISSUE #11418: oc new-app --list fails with ImageStreamImport error
```
# oc new-app --list
error: can't lookup images: ImageStreamImport "newapp" is invalid: spec.images[0].from.name: Invalid value: "*": invalid reference format
```

##### Version
Today's git, f6bd74b925995c46b30a59819586116066c9ba6c.

##### Steps To Reproduce
1. Run oc new-app --list --loglevel=8

##### Current Result
```
...
dockerimagelookup.go:79] checking remote registry for "*"
request.go:558] Request Body: "{\"kind\":\"ImageStreamImport\",\"apiVersion\":\"v1\",\"metadata\":{\"name\":\"newapp\",\"creationTimestamp\"
:null},\"spec\":{\"import\":false,\"images\":[{\"from\":{\"kind\":\"DockerImage\",\"name\":\"*\"},\"importPolicy\":{}}]},\"status\":{}}\n"
round_trippers.go:296] POST https://192.168.121.84:8443/oapi/v1/namespaces/default/imagestreamimports
round_trippers.go:303] Request Headers:
round_trippers.go:306]     User-Agent: oc/v1.4.0 (linux/amd64) openshift/bf5a7d3
round_trippers.go:306]     Accept: application/json, */*
round_trippers.go:306]     Content-Type: application/json
round_trippers.go:321] Response Status: 422 status code 422 in 5 milliseconds
round_trippers.go:324] Response Headers:
round_trippers.go:327]     Cache-Control: no-store
round_trippers.go:327]     Content-Type: application/json
round_trippers.go:327]     Date: Tue, 18 Oct 2016 16:36:47 GMT
round_trippers.go:327]     Content-Length: 416
request.go:908] Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"ImageStreamImport \"newapp\" i
s invalid: spec.images[0].from.name: Invalid value: \"*\": invalid reference format","reason":"Invalid","details":{"name":"newapp","kind":"ImageStreamImport","causes":[{"
reason":"FieldValueInvalid","message":"Invalid value: \"*\": invalid reference format","field":"spec.images[0].from.name"}]},"code":422}
dockerimagelookup.go:98] checking local Docker daemon for "*"
helpers.go:115] error: can't lookup images: ImageStreamImport "newapp" is invalid: spec.images[0].from.name: Invalid value: "*": invalid ref
erence format
```

##### Expected Result
No error.

##### Additional Information
Happens both when there are no ImageStreams/Templates and when there are some.

It's probably a good idea to add a test that runs `oc new-app --list` as part of the fix.

### PR DESCRIPTION
**Pull request by legionus**
Fix #11418 

@miminar @soltysh please review.

### CODE REVIEW

**Comment by legionus** [comment]
[test]

**Comment by legionus** [comment]
[test]

**Comment by legionus** [comment]
[test]

**Comment by pweil-** [comment]
re[test]

**Comment by mfojtik** [comment]
@legionus the failure you got is https://github.com/openshift/origin/issues/11240

**Comment by legionus** [comment]
> @legionus the failure you got is #11240

test_pull_requests_origin_networking fell not because of end-to-end.

https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_networking/262/console

**Comment by dobbymoodge** [comment]
re-re-[test]

File: pkg/generate/app/dockerimagelookup.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func (r DockerClientSearcher) Search(precise bool, terms ...string) (ComponentMa
 		termMatches := ScoredComponentMatches{}
 
 		// first look for the image in the remote docker registry
-		if r.RegistrySearcher != nil {
+		if r.RegistrySearcher != nil && ref.Name != "*" {
```
Comment (miminar) [inline pkg/generate/app/dockerimagelookup.go]: Can you compare against whole `ref.String()` or just `term` instead?

Comment (legionus) [inline pkg/generate/app/dockerimagelookup.go]: OK. I will use `ref.String()`. I would like to emphasize the unacceptability of such values in `ref`.

Comment (legionus) [inline pkg/generate/app/dockerimagelookup.go]: @miminar FIxed

**Comment by legionus** [comment]
@mfojtik merge please

**Comment by mfojtik** [comment]
[merge]

**Comment by mmilata** [comment]
@legionus thanks for fixing this!

**Comment by smarterclayton** [comment]
This also needs to be fixed on the server side - we previously accepted this value from older OC clients and did not return an error, but now we do.  Tightening validation is an API change, and in general we have to be somewhat flexible to old output.

We need to special case this validation on the server and allow it through (and maybe a few others) so that `oc new-app -L` from a v1.3 client works against a v1.4 server.  This is a release blocker since it regresses the API.

**Comment by smarterclayton** [comment]
Please open a follow up issue to track this.

**Pull request merged**